ocfl-java-core/src/main/java/edu/wisc/library/ocfl/core/extension/storage/layout/FlatOmitPrefixLayoutExtension.java

```diff
@@ -0,0 +1,129 @@
+
+/**
+ * Implementation of the <a href="https://ocfl.github.io/extensions/0006-flat-omit-prefix-storage-layout.html">
+ * Flat Omit Prefix Storage Layout</a> extension.
+ *
+ * @author awoods
+ * @since 2021-06-22
+ */
+public class FlatOmitPrefixLayoutExtension implements OcflStorageLayoutExtension {
+
+    public static final String EXTENSION_NAME = "0006-flat-omit-prefix-storage-layout";
+
+    private FlatOmitPrefixLayoutConfig config;
+    private boolean isCaseSensitive = true; // The spec does not provide for configuring this
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getExtensionName() {
+        return EXTENSION_NAME;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDescription() {
+        return "The OCFL object identifiers are expected to contain prefixes which are removed in the mapping to " +
+                "directory names. The OCFL object identifier prefix is defined as all characters before and including a " +
+                "configurable delimiter.";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void init(OcflExtensionConfig config) {
+        // Only set this.config if it is uninitialized
+        if (this.config == null) {
+
+            // Is arg config null?
+            if (config == null) {
+                throw new IllegalArgumentException("Arg config must not be null!");
+            }
+
+            if (!(config instanceof FlatOmitPrefixLayoutConfig)) {
+                throw new OcflExtensionException(String.format("This extension only supports %s configuration. Received: %s",
+                        getExtensionConfigClass(), config));
+            }
+
+            FlatOmitPrefixLayoutConfig castConfig = (FlatOmitPrefixLayoutConfig) config;
+
+            validateConfig(castConfig);
+            this.config = castConfig;
+        }
+    }
+
+    private static void validateConfig(FlatOmitPrefixLayoutConfig config) {
+        if (config != null) {
+            String delimiter = config.getDelimiter();
+            if (delimiter == null || delimiter.isEmpty()) {
+                throw new OcflExtensionException("Digest configuration must not be empty!");
+            }
+        }
+    }
+
+    @Override
+    public Class<? extends OcflExtensionConfig> getExtensionConfigClass() {
+        return FlatOmitPrefixLayoutConfig.class;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String mapObjectId(String objectId) {
+        if (config == null) {
+            throw new OcflExtensionException("This extension must be initialized before it can be used.");
+        }
+
+        // Use lowercase of delimiter and objectId if configured for case-sensitivity (default)
+        String delim = config.getDelimiter();
+        String id = objectId;
+        if (isCaseSensitive) {
+            delim = delim.toLowerCase();
```

Comment:
If you stick with `isCaseSensitive`, then you may was well compute this lower case delimiter value once on `init`.